pkg/objstore/objstore.go

```diff
@@ -119,6 +121,39 @@ func ApplyIterOptions(options ...IterOption) IterParams {
 	return out
 }
 
+// DownloadOption configures the provided params.
+type DownloadDirOption func(params *DownloadDirParams)
+
+// DownloadParams holds the Download() parameters and is used by objstore clients implementations.
+type DownloadDirParams struct {
+	concurrency  int
+	ignoredPaths []string
+}
+
+// WithDownloadIgnoredPaths is an option to set the paths to not be downloaded.
+func WithDownloadIgnoredPaths(ignoredPaths ...string) DownloadDirOption {
+	return func(params *DownloadDirParams) {
+		params.ignoredPaths = ignoredPaths
+	}
+}
+
+// WithFetchConcurrency is an option to set the concurrency of the download operation.
+func WithFetchConcurrency(concurrency int) DownloadDirOption {
+	return func(params *DownloadDirParams) {
+		params.concurrency = concurrency
+	}
+}
+
+func ApplyDownloadOptions(options ...DownloadDirOption) DownloadDirParams {
```

Comment:
> If it is just downloading chunks, I feel we don't need the concurrency limit. The worker limit thing I meant is mainly on another change you did. For this change, overall I think is good. Any thoughts from other maintainers?

Oh.. but downloading chunks in parallel can have the same effect of downloading blocks  in parallel  with regards to use  lots of bandwidth - that 's why added the concurrency limit.


> Let's make it private?

Yeah.. lemme change that.

Reply:
Yes, I agree. My initial thought was to limit on that code only so that the function signature remains unchanged. This one is also good and even more granular, only signature changed a little bit. Let's see what other maintainers think.